proposals/PROPOSALS.md

```diff
@@ -1,141 +1,127 @@
+Discussing alternatives to the proposal or parts of it that may better achieve its goals is relevant, as is questioning whether the goals are in fact beneficial to the community. Promoting alternative goals or proposing ways to reach them is not. In other words, comments like "you should solve this other problem instead" are not encouraged, while comments like "why not use this other library to achieve the goals?" are.
 
-A HFTP is submitted as a pull request against [the official Haskell tech proposal
-repo](https://github.com/haskellfoundation/tech-proposals). Within a week of receiving the pull request, members of the HFTT will acknowledge your submission, validate that it conforms to the proposal template guidelines (see: [TEMPLATE.md](TEMPLATE.md)) and provide feedback to improve the overall quality of the document (if necessary). When the document conforms to the template guidelines, it is ready for formal evaluation by the HFTT members, as well as the broader community.
+Respectful discussion lives up to the [respectful communication guidelines](https://haskell.foundation/guidelines-for-respectful-communication/). Additionally, we must take special effort to listen to each others' experiences and take them seriously. Haskell is used in a variety of contexts with a variety of cultures, interests, and requirements, and it's important that we are able to understand these needs, even if we end up not being able to serve all of them. The needs of researchers, business application developers, and open-source contributors are all important, as are those of people in different industries, different countries, and with different backgrounds.
 
-### Formal Review (up to 5 iterations)
+Productive discussion contributes new insights as it progresses, and doesn't go in circles. When a point has been made, there's usually no need to make it again, although it can be worth checking whether a particular need or interest is unique or shared.
 
-While the majority of general technical commentary and feedback will occur on the proposal pull request and its associated issue, it's hard to tell when a particular proposal is finalized. This what we call "Formal Review".  The entire community is strongly encouraged to comment on, and help improve, a proposal. Ultimately, however, the Foundation needs a mechanism to make a decision, to accept, reject, or push back a proposal. This process is called "Formal Review", and is carried out by the HFTT working group.
+From time to time, the committee will summarize the current state of knowledge in the discussion thread. When there is no longer productive discussion, the committee will proceed to the next step, recommendation.
```

Comment:
I really like these guidelines: actively curating good discussions is hard and important work.

Haskell has a poor track record at being able to keep conversations relevant and forward moving. I just had a fresh idea that may be worth experimenting with. It might be a terrible idea, but I'll throw it out anyway: GitHub allows maintainers of a repo to edit others' comments. Using this feature, I wonder about trying the following:

When a contributor makes (part of) a comment that is unhelpful, a member of the committee edits the comment to say so, with a brief justification. For example, on a proposal for a developing a new JavaScript backend, I might say

> This would dovetail nicely with new special syntax for allowing inline HTML in Haskell code. Specifically, we can imagine defining `<` used in prefix form (that is, with a space before it but not after it) as starting an HTML tag. We can do similar with `>` and `/>` and then add new abilities to the parser to deal with (X)HTML. Specifically, here is the new fragment of the parser: ...

Such a comment is distracting in the context of a discussion of backends, and might get other contributors to weigh in on the idea of doing this new surface-language feature. Accordingly, the committee could edit my comment to include

> **TWG:** The following comment proposes a new idea that might distract us from the content at hand. We encourage other contributors to ignore the following or to move this conversation to a different place.

Note the `TWG:` intro: the power to edit comments must not be abused to change what others have said.

The reason I think this kind of editing might be helpful is that, as a reader of a forum, it's hard for me to know what comments are considered relevant or not. A comment from a moderator several comments down clarifying the situation is too late -- I might have already composed my response and then be unwilling to delete my work. On the other hand, actively labeling distractor comments might help.

Again, maybe this is a terrible idea. But perhaps sharing it inspires others.

Reply:
OK. Sounds like you're saying "terrible idea". I withdraw (completely unoffended!).